R/covariancecorrelation.R

```diff
@@ -1,23 +1,58 @@
 # Compute a matrix containing the pairwise weighted correlations between each column in
-# 'data'.
-weightedPartialCovarianceMatrix <- function(data, weights, correlation = FALSE)
+# 'X'.
+weightedPartialCovarianceMatrix <- function(X, weights, correlation = FALSE)
 {
-    num.cols <- ncol(data)
+    # Removing missing or inappropriate weights.
+    valid <- !is.na(weights) & weights > 0
+    weights <- weights[valid]
+    X <- X[valid, ]
+    # Remove entirely missing rows
+    if (any(all.missing <- apply(X, 1, function(x) all(is.na(x)))))
+    {
+        not.all.missing <- !all.missing
+        X <- X[not.all.missing, ]
+        weights <- weights[not.all.missing]
+    }
+    # Use fast algorithms if no remaining missing values in X
+    if (!anyNA(X))
+    {
```

Comment:
Can you move this block of code to a separate function please?

Reply:
Should be fixed in https://github.com/Displayr/flipStatistics/commit/f2b3e132965add46e9c25543f4195abb8832c480